### PR TITLE
Update security learn pages for SwanLake Alpha3

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
This PR updates the `authentication and authorization` learn page and `the network in the language` page for Swan Lake Alpha3.

Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/584

## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: learn/user-guide/security/authentication-and-authorization.md [hunk 1]
```diff
@@ -27,16 +27,16 @@ The Ballerina HTTP listener can be configured to authenticate and authorize the
...
+The following example represents how a service can be secured. The `http:ServiceConfig` annotation should have an `auth` field, which is an array of elements consisting  `http:FileUserStoreConfigWithScopes`, `http:LdapUserStoreConfigWithScopes`, `http:JwtValidatorConfigWithScopes`, or `http:OAuth2IntrospectionConfigWithScopes` records. Each of these records consists of a record specific configuration (`http:FileUserStoreConfig`, `http:LdapUserStoreConfig`, `http:JwtValidatorConfig`, `http:OAuth2IntrospectionConfig` in this order) and an optional field, which consists of a `string` or `string[]`. The record-specific configuration is used for authentication and the optional field can be used for authorization.
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:30]:
```suggestion
The example below represents how a service can be secured. The `http:ServiceConfig` annotation should have an `auth` field, which is an array of elements consisting of   `http:FileUserStoreConfigWithScopes`, `http:LdapUserStoreConfigWithScopes`, `http:JwtValidatorConfigWithScopes`, or `http:OAuth2IntrospectionConfigWithScopes` records. Each of these records consists of a record specific configuration (`http:FileUserStoreConfig`, `http:LdapUserStoreConfig`, `http:JwtValidatorConfig`, `http:OAuth2IntrospectionConfig` in this order) and an optional field, which consists of a `string` or `string[]`. The record-specific configuration is used for authentication and the optional field can be used for authorization.
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 2]
```diff
@@ -88,6 +88,112 @@ Also, the security enforcement that is done for the service using the `http:Serv
 
 #### Basic Auth
 
+##### File User Store
+
+Ballerina supports File user store Basic Authentication and Authorization for services/resources. The `auth` field of a service/resource annotation should have a `http:FileUserStoreConfigWithScopes` record as an element. If the `fileUserStoreConfig` field is assigned with the `http:FileUserStoreConfig` implementation, the authentication will be evaluated. Optionally, you can have the `string|string[]` value for the `scopes` field also. Then, the authorization will be evaluated.
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:93]:
```suggestion
Ballerina supports the file user store basic authentication and authorization for services/resources. The `auth` field of a service/resource annotation should have an `http:FileUserStoreConfigWithScopes` record as an element. If the `fileUserStoreConfig` field is assigned with the `http:FileUserStoreConfig` implementation, the authentication will be evaluated. Optionally, you can have the `string|string[]` value for the `scopes` field also. Then, the authorization will be evaluated.
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 3]
```diff
@@ -88,6 +88,112 @@ Also, the security enforcement that is done for the service using the `http:Serv
...
+
+The `http:FileUserStoreConfig` configurations is kept blank for future improvements and backward compatibility.
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:95]:
```suggestion
The `http:FileUserStoreConfig` configurations are kept blank for future improvements and backward compatibility.
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 4]
```diff
@@ -88,6 +88,112 @@ Also, the security enforcement that is done for the service using the `http:Serv
...
+
+The file user store is defined in `Config.toml` as follows:
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:97]:
```suggestion
The file user store is defined in the `Config.toml` as follows:
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 5]
```diff
@@ -88,6 +88,112 @@ Also, the security enforcement that is done for the service using the `http:Serv
...
+    ]
+}
+service /foo on securedEP {
+    resource function get bar() returns string {
+        return "Hello, World!";
+    }
+}
+```
+
+When the service is invoked without authentication information or invalid authentication information, an authentication failure will occur:
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:141]:
```suggestion
When the service is invoked without authentication information or invalid authentication information, an authentication failure will occur.
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 6]
```diff
@@ -111,18 +217,19 @@ The `http:LdapUserStoreConfig` configurations include:
 * `membershipAttribute` - Define the attribute that contains the distinguished names (DN) of user objects that are in a group
 * `userRolesCacheEnabled` -  To indicate whether to cache the role list of a user
 * `connectionPoolingEnabled` - Define whether LDAP connection pooling is enabled
-* `connectionTimeoutInMillis` - Timeout in making the initial LDAP connection
-* `readTimeoutInMillis` -  Read timeout in milliseconds for LDAP operations
-* `secureClientSocket` - The SSL configurations for the LDAP client socket. This needs to be configured in order to communicate through LDAPs
+* `connectionTimeout` - Connection timeout (in seconds) in making the initial LDAP connection
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:220]:
```suggestion
* `connectionTimeout` - Connection timeout (in seconds) when making the initial LDAP connection
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 7]
```diff
@@ -111,18 +217,19 @@ The `http:LdapUserStoreConfig` configurations include:
...
+* `readTimeout` - Reading timeout (in seconds) for LDAP operations
+* `secureSocket` - The SSL configurations for the LDAP client socket. This needs to be configured in order to communicate through LDAPs
+    * `cert` - Configurations associated with `crypto:TrustStore` or single certificate file that the client trusts
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:223]:
```suggestion
    * `cert` - Configurations associated with the `crypto:TrustStore` or single certificate file that the client trusts
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 8]
```diff
@@ -226,26 +333,26 @@ The `http:JwtValidatorConfig` configurations include:
...
-    * `url` - URL of the JWKs endpoint
-    * `cacheConfig` - Configurations related to the cache, which are used to store preloaded JWKs information
-    * `clientConfig` - HTTP client configurations, which call the JWKs endpoint
-* `cacheConfig` - Configurations related to the cache, which are used to store the parsed JWT information
-
-> **Note:** For demonstration purposes, the `ballerinaTruststore.p12` included with the Ballerina runtime is used. In a production deployment, the truststore should only contain the public key certificates of the trusted JWT issuers.
+* `clockSkew` - Clock skew (in seconds) that can be used to avoid token validation failures due to clock synchronization problems
+* `signatureConfig` - JWT signature configurations
+    * `jwksConfig` - JWKs configurations
+        * `url` - URL of the JWKs endpoint
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:339]:
```suggestion
        * `url` - URL of the JWK endpoint
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 9]
```diff
@@ -226,26 +333,26 @@ The `http:JwtValidatorConfig` configurations include:
...
+        * `cacheConfig` - Configurations related to the cache, which are used to store preloaded JWKs information
+        * `clientConfig` - HTTP client configurations, which call the JWKs endpoint
+    * `certFile` - Public certificate file
+    * `trustStoreConfig` - JWT trust store configurations
+        * `trustStore` - Trust store used for signature verification
+        * `certificateAlias` - Token-signed public key certificate alias
+* `cacheConfig` - Configurations related to the cache used to store parsed JWT information
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:346]:
```suggestion
* `cacheConfig` - Configurations related to the cache, which are used to store parsed JWT information
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 10]
```diff
@@ -342,17 +445,26 @@ The `http:OAuth2IntrospectionConfig` configurations include:
 * `tokenTypeHint` - A hint about the type of the token submitted for introspection
 * `optionalParams` - Map of optional parameters used for the introspection endpoint
 * `cacheConfig` - Configurations for the cache used to store the OAuth2 token and other related information
-* `defaultTokenExpTimeInSeconds` - Expiration time of the tokens if introspection response does not contain an `exp` field
+* `defaultTokenExpTime` - Expiration time (in seconds) of the tokens if introspection response does not contain an `exp` field
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:448]:
```suggestion
* `defaultTokenExpTime` - Expiration time (in seconds) of the tokens if the introspection response does not contain an `exp` field
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 11]
```diff
@@ -342,17 +445,26 @@ The `http:OAuth2IntrospectionConfig` configurations include:
...
 * `clientConfig` - HTTP client configurations, which call the introspection server
+    * `httpVersion` - The HTTP version of the client
+    * `customHeaders` - The list of custom HTTP headers
+    * `customPayload` - The list of custom HTTP payload parameters
+    * `auth` - The client auth configurations
+        * `oauth2:ClientCredentialsGrantConfig`|`oauth2:PasswordGrantConfig`|`oauth2:DirectTokenConfig`
+    * `secureSocket` - SSL/TLS related configurations
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:455]:
```suggestion
    * `secureSocket` - SSL/TLS-related configurations
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 12]
```diff
@@ -342,17 +445,26 @@ The `http:OAuth2IntrospectionConfig` configurations include:
...
+        * `disable` - Disable SSL validation
+        * `cert` - Configurations associated with `crypto:TrustStore` or single certificate file that the client trusts
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:457]:
```suggestion
        * `cert` - Configurations associated with the `crypto:TrustStore` or single certificate file that the client trusts
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 13]
```diff
@@ -342,17 +445,26 @@ The `http:OAuth2IntrospectionConfig` configurations include:
...
+        * `key` - Configurations associated with `crypto:KeyStore` or combination of certificate and private key of the client
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:458]:
```suggestion
        * `key` - Configurations associated with the `crypto:KeyStore` or a combination of the certificate and private key of the client
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 14]
```diff
@@ -516,18 +619,23 @@ The `http:JwtIssuerConfig` configurations include:
 * `username` - JWT username, which is mapped to `sub`
 * `issuer` - JWT issuer, which is mapped to `iss`
 * `audience` - JWT audience, which is mapped to `aud`
-* `keyId` - JWT key ID, which is mapped to `kid`
+* `jwtId` - JWT ID, which is mapped to `jti`
+* `keyId` - JWT key ID, which is mapped `kid`
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:623]:
```suggestion
* `keyId` - JWT key ID, which is mapped to the `kid`
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 15]
```diff
@@ -516,18 +619,23 @@ The `http:JwtIssuerConfig` configurations include:
 * `username` - JWT username, which is mapped to `sub`
 * `issuer` - JWT issuer, which is mapped to `iss`
 * `audience` - JWT audience, which is mapped to `aud`
-* `keyId` - JWT key ID, which is mapped to `kid`
+* `jwtId` - JWT ID, which is mapped to `jti`
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:622]:
```suggestion
* `jwtId` - JWT ID, which is mapped to the `jti`
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 16]
```diff
@@ -516,18 +619,23 @@ The `http:JwtIssuerConfig` configurations include:
...
-    * `keyAlias` - Signing key alias
-    * `keyPassword` - Signing key password
+* `expTime` - Expiry time in seconds
+* `signatureConfig` - JWT signature configurations
+    * `algorithm` - Cryptographic signing algorithm for JWS
+        * `jwt:RS256` - The RSA-SHA256 algorithm
+        * `jwt:RS384` - The RSA-SHA384 algorithm
+        * `jwt:RS512` - The RSA-SHA512 algorithm
+        * `jwt:NONE` - Unsecured JWTs (no signing)
+    * `config` - Key store configurations or private key configurations
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:632]:
```suggestion
    * `config` - Keystore configurations or private key configurations
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 17]
```diff
@@ -516,18 +619,23 @@ The `http:JwtIssuerConfig` configurations include:
...
+        * `keyStore` - Keystore to be used in JWT signing
+        * `keyAlias` - Signing key alias
+        * `keyPassword` - Signing key password
+            * --- OR ---
+        * `keyFile` - Private key to be used in JWT signing
+        * `keyPassword` - Password of the private key, if encrypted
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:638]:
```suggestion
        * `keyPassword` - Password of the private key 
        (if encrypted)
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 18]
```diff
@@ -617,13 +718,22 @@ The `http:OAuth2ClientCredentialsGrantConfig` configurations include:
 * `clientId` - Client ID for the client credentials grant authentication
 * `clientSecret` - Client secret for the client credentials grant authentication
 * `scopes` - Scope(s) of the access request
-* `defaultTokenExpInSeconds` - Expiration time of the tokens if the authorization server response does not contain an `expires_in` field
-* `clockSkewInSeconds` - Clock skew in seconds that can be used to avoid token validation failures due to clock synchronization problems
+* `defaultTokenExpTime` - Expiration time (in seconds) of the tokens if authorization server response does not contain an `expires_in` field
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:721]:
```suggestion
* `defaultTokenExpTime` - Expiration time (in seconds) of the tokens if the authorization server response does not contain an `expires_in` field
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 19]
```diff
@@ -617,13 +718,22 @@ The `http:OAuth2ClientCredentialsGrantConfig` configurations include:
...
 * `credentialBearer` - Bearer of the authentication credentials, which is sent to the authorization endpoint
     * `http:AUTH_HEADER_BEARER` - Indicates that the authentication credentials should be sent via the Authentication Header
     * `http:POST_BODY_BEARER` - Indicates that the Authentication credentials should be sent via the body of the POST request
 * `clientConfig` - HTTP client configurations, which are used to call the authorization endpoint
+    * `httpVersion` - The HTTP version of the client
+    * `customHeaders` - The list of custom HTTP headers
+    * `customPayload` - The list of custom HTTP payload parameters
+    * `auth` - The client auth configurations
+        * `oauth2:ClientCredentialsGrantConfig`|`oauth2:PasswordGrantConfig`|`oauth2:DirectTokenConfig`
+    * `secureSocket` - SSL/TLS related configurations
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:733]:
```suggestion
    * `secureSocket` - SSL/TLS-related configurations
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 20]
```diff
@@ -617,13 +718,22 @@ The `http:OAuth2ClientCredentialsGrantConfig` configurations include:
...
+        * `disable` - Disable SSL validation
+        * `cert` - Configurations associated with `crypto:TrustStore` or single certificate file that the client trusts
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:735]:
```suggestion
        * `cert` - Configurations associated with the `crypto:TrustStore` or single certificate file that the client trusts
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 21]
```diff
@@ -617,13 +718,22 @@ The `http:OAuth2ClientCredentialsGrantConfig` configurations include:
...
+        * `key` - Configurations associated with `crypto:KeyStore` or combination of certificate and private key of the client
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:736]:
```suggestion
        * `key` - Configurations associated with the `crypto:KeyStore` or a combination of the certificate and private key of the client
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 22]
```diff
@@ -681,13 +785,22 @@ The `http:OAuth2PasswordGrantConfig` configurations include:
         * `http:AUTH_HEADER_BEARER` - Indicates that the authentication credentials should be sent via the Authentication Header
         * `http:POST_BODY_BEARER` - Indicates that the Authentication credentials should be sent via the body of the POST request
     * `clientConfig` - HTTP client configurations, which are used to call the authorization endpoint
-* `defaultTokenExpInSeconds` - Expiration time of the tokens if the authorization server response does not contain an `expires_in` field
-* `clockSkewInSeconds` - Clock skew in seconds that can be used to avoid token validation failures due to clock synchronization problems
+* `defaultTokenExpTime` - Expiration time (in seconds) of the tokens if authorization server response does not contain an `expires_in` field
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:788]:
```suggestion
* `defaultTokenExpTime` - Expiration time (in seconds) of the tokens if the authorization server response does not contain an `expires_in` field
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 23]
```diff
@@ -681,13 +785,22 @@ The `http:OAuth2PasswordGrantConfig` configurations include:
...
 * `credentialBearer` - Bearer of the authentication credentials, which is sent to the authorization endpoint
     * `http:AUTH_HEADER_BEARER` - Indicates that the authentication credentials should be sent via the Authentication Header
     * `http:POST_BODY_BEARER` - Indicates that the Authentication credentials should be sent via the body of the POST request
 * `clientConfig` - HTTP client configurations, which are used to call the authorization endpoint
+    * `httpVersion` - The HTTP version of the client
+    * `customHeaders` - The list of custom HTTP headers
+    * `customPayload` - The list of custom HTTP payload parameters
+    * `auth` - The client auth configurations
+        * `oauth2:ClientCredentialsGrantConfig`|`oauth2:PasswordGrantConfig`|`oauth2:DirectTokenConfig`
+    * `secureSocket` - SSL/TLS related configurations
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:800]:
```suggestion
    * `secureSocket` - SSL/TLS-related configurations
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 24]
```diff
@@ -681,13 +785,22 @@ The `http:OAuth2PasswordGrantConfig` configurations include:
...
+        * `disable` - Disable SSL validation
+        * `cert` - Configurations associated with `crypto:TrustStore` or single certificate file that the client trusts
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:802]:
```suggestion
        * `cert` - Configurations associated with the `crypto:TrustStore` or single certificate file that the client trusts
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 25]
```diff
@@ -681,13 +785,22 @@ The `http:OAuth2PasswordGrantConfig` configurations include:
...
+        * `key` - Configurations associated with `crypto:KeyStore` or combination of certificate and private key of the client
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:803]:
```suggestion
        * `key` - Configurations associated with the `crypto:KeyStore` or a combination of the certificate and private key of the client
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 26]
```diff
@@ -750,13 +854,22 @@ The `http:OAuth2DirectTokenConfig` configurations include:
 * `clientId` - Client ID for authentication against the authorization endpoint
 * `clientSecret` - Client secret for authentication against the authorization endpoint
 * `scopes` - Scope(s) of the access request
-* `defaultTokenExpInSeconds` - Expiration time of the tokens if authorization server response does not contain an `expires_in` field
-* `clockSkewInSeconds` - Clock skew in seconds that can be used to avoid token validation failures due to clock synchronization problems
+* `defaultTokenExpTime` - Expiration time (in seconds) of the tokens if authorization server response does not contain an `expires_in` field
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:857]:
```suggestion
* `defaultTokenExpTime` - Expiration time (in seconds) of the tokens if the authorization server response does not contain an `expires_in` field
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 27]
```diff
@@ -750,13 +854,22 @@ The `http:OAuth2DirectTokenConfig` configurations include:
...
 * `credentialBearer` - Bearer of the authentication credentials, which is sent to the authorization endpoint
     * `http:AUTH_HEADER_BEARER` - Indicates that the authentication credentials should be sent via the Authentication Header
     * `http:POST_BODY_BEARER` - Indicates that the Authentication credentials should be sent via the body of the POST request
 * `clientConfig` - HTTP client configurations, which are used to call the authorization endpoint
+    * `httpVersion` - The HTTP version of the client
+    * `customHeaders` - The list of custom HTTP headers
+    * `customPayload` - The list of custom HTTP payload parameters
+    * `auth` - The client auth configurations
+        * `oauth2:ClientCredentialsGrantConfig`|`oauth2:PasswordGrantConfig`|`oauth2:DirectTokenConfig`
+    * `secureSocket` - SSL/TLS related configurations
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:869]:
```suggestion
    * `secureSocket` - SSL/TLS-related configurations
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 28]
```diff
@@ -750,13 +854,22 @@ The `http:OAuth2DirectTokenConfig` configurations include:
...
+        * `disable` - Disable SSL validation
+        * `cert` - Configurations associated with `crypto:TrustStore` or single certificate file that the client trusts
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:871]:
```suggestion
        * `cert` - Configurations associated with the `crypto:TrustStore` or single certificate file that the client trusts
```

File: learn/user-guide/security/authentication-and-authorization.md [hunk 29]
```diff
@@ -750,13 +854,22 @@ The `http:OAuth2DirectTokenConfig` configurations include:
...
+        * `key` - Configurations associated with `crypto:KeyStore` or combination of certificate and private key of the client
```
Comment (praneesha) [inline learn/user-guide/security/authentication-and-authorization.md:872]:
```suggestion
        * `key` - Configurations associated with the `crypto:KeyStore` or a combination of certificate and private key of the client
```

File: learn/user-guide/security/authentication-and-authorization.md (hunk 8 above)
Comment (ldclakmal) [reply learn/user-guide/security/authentication-and-authorization.md:339]: This is the technical term we used for that endpoint. So let's keep it as it is.

**Review by praneesha — APPROVED** [review]

**Pull request merged**